### PR TITLE
ADS-3428: End-to-End current variation for ev1 requests

### PR DESCRIPTION
**Pull request by manikandan-ravikumar**
This PR extends existing hydrogen app with ability use `useNostoContext` on client pages and `useSession` on server pages. As Hydrogen pages are mostly rendered on server side, it's not possible to rerender changes without reloading the page. Hence, this PR uses context to rerender page whereever possible. Also uses session to reload pages where react rerender is not possible. 

[Related PR](https://github.com/Nosto/nosto-react/pull/20)

## JIRA
[CT-8](https://nostosolutions.atlassian.net/browse/CT-8)

### CODE REVIEW

**Comment by manikandan-ravikumar** [comment]
`nosto_react` package version will be updated after review and release of changes in [this](https://github.com/Nosto/nosto-react/pull/20) PR

File: hydrogen.config.js [hunk 1]
```diff
@@ -7,12 +7,12 @@ import {
 export default defineConfig({
   routes: import.meta.globEager('./src/routes/**/*.server.[jt](s|sx)'),
   shopify: {
-    storeDomain: 'hydrogen-preview.myshopify.com',
-    storefrontToken: '3b580e70970c4528da70c98e097c2fa0',
+    storeDomain: 'maninostolocal.myshopify.com',
+    storefrontToken: '4d26d6886f78c9b22d532c0578497ed1',
```
Comment (dominikgilg) [inline hydrogen.config.js:11]: We need to revert this to the default values

File: src/components/Header.client.jsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import {useEffect, useState} from 'react';
+import {useEffect, useState, useContext} from 'react';
```
Comment (dominikgilg) [inline src/components/Header.client.jsx:1]: Is that necessary? Can't see it used somewhere

File: hydrogen.config.js [hunk 3]
```diff
@@ -7,12 +7,12 @@ import {
...
     storefrontApiVersion: '2022-07',
   },
   nosto: {
-    merchantId: 'NOSTO_MERCHANT_ID',
+    merchantId: 'shopify-61881221345',
```
[the remark this answers is not in the document]
Comment (manikandan-ravikumar) [reply hydrogen.config.js]: Fixed

File: src/components/Header.client.jsx (hunk 2 above)
Comment (manikandan-ravikumar) [reply src/components/Header.client.jsx]: Remove most of unnecessary imports

**Comment by manikandan-ravikumar** [comment]
Implemented under different PR https://github.com/Nosto/nosto-shopify-hydrogen/pull/15

**Pull request closed without merging**